core/module.h

```diff
@@ -291,18 +291,28 @@ class Module {
    * Compute placement constraints based on the current module and all
    * downstream modules (i.e., modules connected to out ports.
    */
-  placement_constraint ComputePlacementConstraints() const;
+  placement_constraint ComputePlacementConstraints(
+      std::unordered_set<const Module *> &visited) const;
```

Comment:
Google style guide requires using a pointer for mutable arguments, and const reference for immutable arguments.